app/components/file/report-btn/component.js

```diff
@@ -0,0 +1,278 @@
+    this.isShowCopyPasswordModal = false;
+    this.isShowGenerateReportModal = false;
+  }
+
+  // Events
+
+  observeReportCounter() {
+    this.getReports.perform();
+  }
+
+  // Methods
+
+  /**
+   * Method to generate a new report
+   */
+  @task(function* () {
+    this.isReportGenerating = true;
+    try {
+      const genReport = yield this.store.createRecord(this.modelName, {
+        fileId: this.fileId
+      });
+      yield genReport.save();
+      yield this.getReports.perform();
+      this.notify.success(this.intl.t('reportIsGettingGenerated'));
+      this.isReportGenerating = false;
+    } catch (error) {
+      this.notify.error(parseError(error), this.intl.t('reportGenerateError'));
+    }
+  }).restartable() generateReport;
+
+  /**
+   * Method to re-generate a new report from security dashboard
+   */
+  @task(function* () {
+    const emails = this.emailsToSend;
+    let data = {};
+    if (!isEmpty(emails)) {
+      data = {
+        emails: emails.split(",").map(item => item.trim())
+      };
+    }
+    const adapter = this.store.adapterFor(this.modelName);
+    try {
+      yield adapter.reGenerateReport(this.modelName, this.fileId, data)
+      this.isReportReGenerated = true;
+      this.emailsToSend = '';
+      this.sentEmailIds = data.emails;
+    } catch (error) {
+      this.notify.error(parseError(error), this.intl.t('pleaseTryAgain'))
+    }
+  }).restartable() reGenerateReport;
+
+  /**
+   * Method to get reports under a file
+   */
+  @task(function* () {
+    const query = {
+      fileId: this.fileId,
+      limit: REPORT.MAX_LIMIT
+    }
+    try {
+      this.reports = yield this.store.query(this.modelName, query);
+    } catch {
+      this.reports = [];
+    }
+  })
+  getReports;
+
+  /**
+   * Method to get/download pdf/type based report
+   */
+  @task(function* (type, reportId) {
+    triggerAnalytics('feature', ENV.csb.reportDownload);
+    const adapter = this.store.adapterFor(this.modelName);
+    try {
+      const report = yield adapter.getReportByType(this.modelName, reportId, type);
+      if (report && report.url) {
+        window.location = report.url;
+        this.isShowCopyPasswordModal = true;
+      } else {
+        this.notify.error(this.intl.t('downloadUrlNotFound'))
+      }
+    } catch {
+      this.notify.error(this.intl.t('reportIsGettingGenerated'))
+    }
+  }).restartable() getReportByType;
+
+  /**
+   * Method to download excel report
+   */
+  @task(function* (type) {
+    const adapter = this.store.adapterFor(this.modelName);
+    try {
+      const data = yield adapter.downloadExternalReport(this.modelName, this.fileId)
+      if (data && data[type.format]) {
+        window.location = data[type.format];
+      } else {
+        this.notify.error(this.intl.t('noReportExists', { format: type.label }));
+      }
+    } catch (error) {
+      this.notify.error(parseError(error, this.intl.t('pleaseTryAgain')))
+    }
+  }).restartable() downloadExternalReport;
```

Comment:
It should not be called as external report - just use report.

Reply:
Somewhere it was referred as external reports, which is why I used the name here. I'll change it 👍